app/services/remote_identities/create_service.rb

```diff
@@ -32,21 +32,25 @@ module RemoteIdentities
   class CreateService
     attr_reader :user, :model
 
-    def self.call(user:, integration:, token:)
-      new(user:, integration:, token:).call
+    def self.call(user:, integration:, token:, force_update: false)
+      new(user:, integration:, token:, force_update:).call
     end
 
-    def initialize(user:, integration:, token:)
+    def initialize(user:, integration:, token:, force_update: false)
       @user = user
       @integration = integration
       @token = token
+      @force_update = force_update
 
       @model = RemoteIdentity.find_or_initialize_by(user:, auth_source: token.auth_source, integration:)
     end
 
     def call
-      @model.origin_user_id = @integration.extract_origin_user_id(@token)
-      if @model.changed?
+      if @model.new_record? || @force_update
+        user_id = @integration.extract_origin_user_id(@token)
+        return user_id if user_id.failure?
+
+        @model.origin_user_id = user_id.result
```

Comment:
Would it make sense to check if model changed to avoid unnecessary event emission? 
```suggestion
        @model.origin_user_id = user_id.result
        return ServiceResult.success(result: @model) unless @model.changed?
```

Reply:
done

Reply:
Thanks!